lib/recipe/bloc/recipe_bloc.dart

```diff
@@ -546,14 +569,19 @@ class RecipeBloc extends Bloc<RecipeEvent, RecipeState> {
         List.from(recipes[changedRecipeIndex].ingredients);
     int changedIngredientIndex = ingredients.indexOf(event.ingredient);
 
-    Ingredient ingredient = ingredients[changedIngredientIndex]
-        .copyWith(name: event.ingredientName, saved: false);
+    String ingredientName = event.ingredientName;
+    Map<String, dynamic> ingredientMap = json.decode(state.ingredientsJson!);
+
+    Ingredient ingredient = ingredients[changedIngredientIndex].copyWith(
+        name: ingredientName,
+        foodComposition: ingredientMap[ingredientName],
+        saved: false);
```

Comment:
Was there a change in the event such that now it holds ingredient name?

Reply:
It now holds the ingredients food composition information as a start to a future implementation of dietary analysis from within the app. This can of course be removed and only implemented when in-app dietary evaluation is eventually implemented